driver/normalizer/strconv_test.go

```diff
@@ -0,0 +1,78 @@
+package normalizer
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var testCasesUnquote = []struct {
+	quoted   string
+	unquoted string
+	// If this is non-empty it means that quoteing back unqoted string does not
```

Comment:
This helps, thanks, I think I understand better now. With that in mind, I would like to suggest a further clarification (which you should feel free to ignore if it's wrong):

```go
// In some cases unquoting and then re-quoting a quoted string does not produce a
// string that is bitwise identical to the original, even though they denote the same bytes.
// This can happen, for example, if we switch between hex and octal encoding of a byte.
// Test cases where this happens should set canonicalUnquoted to a non-empty string that 
// that decodes via Go's native rules to the byte sequence we want.
// Otherwise (if canonicalUnquoted == "") the test assumes the encoding should round-trip.
```

What do you think?

Reply:
Makes me wonder if we shouldn't just include this in _all_ the test cases just for symmetry—even the ones where it's not strictly necessary.

Reply:
Sounds much better to me 🎉 And that also would simplify the test logic.
Addressed in 7f92d74